projects/plutus/src/ltc_afe.c

```diff
@@ -0,0 +1,286 @@
+#include "crc15.h"
+#include "delay.h"
+#include "ltc_afe.h"
+#include "ltc68041.h"
+#include "plutus_config.h"
+
+static bool s_discharging_cells[LTC6804_CELLS_PER_DEVICE * PLUTUS_AFE_DEVICES_IN_CHAIN] = { false };
+
+static uint16_t s_read_reg_cmd[NUM_LTC_AFE_REGISTER] = {
+  LTC6804_RDCFG_RESERVED,
+  LTC6804_RDCVA_RESERVED,
+  LTC6804_RDCVB_RESERVED,
+  LTC6804_RDCVC_RESERVED,
+  LTC6804_RDCVD_RESERVED,
+  LTC6804_RDAUXA_RESERVED,
+  LTC6804_RDAUXA_RESERVED,
+  LTC6804_RDSTATA_RESERVED,
+  LTC6804_RDSTATB_RESERVED,
+  LTC6804_RDCOMM_RESERVED
+};
+
+static uint8_t s_voltage_reg[NUM_LTC_AFE_VOLTAGE_REGISTER] = {
+  LTC_AFE_REGISTER_CELL_VOLTAGE_A,
+  LTC_AFE_REGISTER_CELL_VOLTAGE_B,
+  LTC_AFE_REGISTER_CELL_VOLTAGE_C,
+  LTC_AFE_REGISTER_CELL_VOLTAGE_D
+};
+
+static void prv_wakeup_idle(const LTCAFESettings *afe) {
+  gpio_set_pin_state(&afe->cs, GPIO_STATE_LOW);
+  delay_us(2);
+  gpio_set_pin_state(&afe->cs, GPIO_STATE_HIGH);
+}
+
+static void prv_build_cmd(uint16_t command, uint8_t *cmd) {
```

Comment:
Recommend cmd len to make sure you don't accidentally pass in an incorrectly sized array.